File: MH/DAY6/9012.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs.readFileSync(filePath).toString().split("\n");
let stack = [];
const T = input.shift();

for (let i = 0; i < T; i++) {
  const arr = input[i].trim().split("");

  for (let j = 0; j < input[i].trim().length; j++) {
    stack.push(arr[j]);
    if (stack.slice(-2).join("") === "()") {
      stack.pop();
      stack.pop();
    }
  }
  console.log(stack.length === 0 ? "YES" : "NO");
  stack = [];
}

File: SY/stack/9012.py
n=int(input())
li=[list(input()) for _ in range(n)]

for i in range(n):
	stack=[]
	for j in li[i]:
		stack.append(j)
		if len(stack)>1:
			if stack[-2]=='(' and stack[-1]==')':
				stack.pop()
				stack.pop()
	if not stack:
		print('YES')
	else:
		print('NO')
